components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -496,62 +400,60 @@ ::update_spa_data_from_file(
           SPAHorizInterp&       spa_horiz_interp,
           SPAInput&             spa_data)
 {
+  start_timer("EAMxx::SPA::update_spa_data_from_file");
```

Comment:
This routine is run once a month and is the possible performance bottleneck, right?

One thing that would help is to add a bunch of timers throughout this routine to assess the cost of each piece and report performance results in a comment. If the slowdown is in something other than reading the data from the file, then there's quite a lot that can be done to speed up the code.

I'm also fine with merging this PR and then doing a separate speedup PR.

Reply:
I'll add more timers for my timing check and report back.

Reply:
@ambrad , @jeff-cohere and @bartgol .  The preliminary timer info is in and it looks pretty bad for the apply_remap call for the 4 banded variables which are 3D.   Here's a taste of timing on summit, ne30pg2

```
"a_i:EAMxx::SPA::update_spa_data_from_file"                -         30       30 6.000000e+01   6.618477e+03   220.617 (    10      0)   220.614 (     1      0)
"a_i:EAMxx::SPA::update_spa_data_from_file::read_data"     -         30       30 6.000000e+01   3.921619e+02    13.074 (    11      0)    13.070 (    19      0)
"a_i:EAMxx::SPA::update_spa_data_from_file::apply_remap"   -         30       30 6.000000e+01   6.055629e+03   207.534 (     4      0)   196.535 (    24      0)
"a_i:EAMxx::GSMap::apply_remap_1d"                         -         30       30 6.000000e+01   1.459733e+00     0.050 (     4      0)     0.047 (    23      0)
"a_i:EAMxx::GSMap::apply_remap_2d"                         -         30       30 6.000000e+01   1.038210e+02     3.574 (     4      0)     3.381 (    24      0)
"a_i:EAMxx::GSMap::apply_remap_3d"                         -         30       30 2.400000e+02   5.949352e+03   203.876 (     4      0)   193.073 (    24      0)
"a_i:EAMxx::SPA::update_spa_data_from_file::copy_and_pad"  -         30       30 6.000000e+01   1.470893e-02     0.001 (    16      0)     0.000 (    21      0)
```

There are 14 swbands and 16 lwbands in the third dimension, and 4 of those in total, so 3*14 + 16 = 58.  So 3D remapping is 58X more work.  So that explains why 3D remapping is around 60X more time.

I'm running more timing so I can compare master.  But I mention this now in case it can spur creative ideas about speeding up this chunk of code.  It is the biggest bottleneck that I can see.

Reply:
Here is current `master`, same run

```
"a_i:EAMxx::SPA::update_spa_data_from_file"               -         30       30 6.000000e+01   4.059888e+02    13.538 (    25      0)    13.531 (     3      0)
"a_i:EAMxx::SPA::update_spa_data_from_file::read_data"    -         30       30 6.000000e+01   3.368266e-01     0.016 (    25      0)     0.009 (     0      0)
"a_i:EAMxx::SPA::update_spa_data_from_file::apply_remap"  -         30       30 6.000000e+01   3.573152e+02    13.512 (     0      0)    10.239 (     1      0)
```

Note, here remapping is one monolithic set of for loops, so there is no breakdown by dimension like above.  Still, looking at more than 10X a slowdown overall.  Current hypothesis is the host/device copies in the new remapper.  This can be improved, so I am going to investigate next.

Reply:
Just to be clear, these timers are for init (a_i prefix => atm init). 1. Do these same timers apply when time stepping? 2. That aside, my sense is that this init time is really bad, perhaps the vast majority of the init time for this case (ne30pg2 on 5 nodes of Summit, I suspect). True?

Reply:
The times I report above are the sum of all times the "update_spa_data" is called.  In this case twice, at initialization.  If I ran for at least a month there would be another 3rd call then to update to the next month of data.   But more to the point, yes, a huge % of init is in this one section.  Total init times are

```
master:
"a_i:EAMxx::init"                                         -         30       30 1.800000e+02   7.822712e+02    26.098 (    10      0)    26.053 (    12      0)
new remap:
"a_i:EAMxx::init"                                          -         30       30 1.800000e+02   6.977105e+03   232.574 (    10      0)   232.555 (     1      0)
```

Just to be clear.  SPA initialization calls the data update twice to get things started.  But updating data is called once per month.

Reply:
Ok, thanks. Staying completely on host, as discussed on slack, seems like the obvious alternative to try.